src/ctia/documentation/routes.clj

```diff
@@ -49,49 +45,53 @@
            :class page-class}
      html-body]]))
 
-(defn render-markdown
-  "render a mardown file into an html webpage"
-  [file]
-  {:status 200
-   :headers {"Content-Type" "text/html"}
-   :body (-> file
-             md-to-html-string
-             decorate-markdown)})
-
-(defn render-default
-  "default render for unknown file types"
-  [file type]
-  {:status 200
-   :headers {"Content-Type" type}
-   :body file})
-
-(defn render
-  "render a file by type"
-  [file type]
-  (condp = type
-    "text/markdown" (render-markdown file)
-    (render-default file type)))
-
 (defn render-request
   "read the requested file from resources, render it if needed"
   [path-info]
-  (let [file-path (subs path-info 1)
-        file-content (get-file-content file-path)
-        file-type (ext-mime-type file-path mime-overrides)]
-
-    (if file-content
-      (render file-content file-type)
+  (or (resource-response (str doc-resource-prefix path-info)
+                         {:allow-symlinks? false})
       {:status 404
-       :body "The requested page couldn't be found."})))
+       :body "The requested page couldn't be found."}))
 
 (def render-request-with-cache
   "request cache wrapper"
   (memo/ttl
    render-request
    :ttl/threshold cache-ttl-ms))
 
+(defn markdown->html [resp]
+  (let [body (-> resp
+                 (get :body)
+                 (slurp)
+                 (md-to-html-string)
+                 (decorate-markdown))
+        body-length (count (.getBytes body "UTF-8"))]
+    (-> resp
+        (assoc :body body)
+        (response/content-type "text/html")
+        (response/charset "UTF-8")
+        (response/update-header "Content-Length" (constantly (str body-length))))))
```

Comment:
```suggestion
        (response/header "Content-Length" (str body-length)))))
```

Reply:
Thanks! No idea how I oversee `header` :) It's even stringifying the content automatically